app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,130 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val _popularSelectedCategory = MutableStateFlow(MovieCategory.POPULAR_STREAMING)
+    private val _nowPlayingSelectedCategory = MutableStateFlow(MovieCategory.NOW_PLAYING_MOVIES)
+    private val _upcomingSelectedCategory = MutableStateFlow(MovieCategory.UPCOMING_TODAY)
+
+    private val _popularCategoryViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val popularCategoryViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularCategoryViewState.asStateFlow()
+    private val _nowPlayingCategoryViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val nowPlayingCategoryViewState: StateFlow<HomeMovieCategoryViewState> =
+        _nowPlayingCategoryViewState.asStateFlow()
+    private val _upcomingCategoryViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val upcomingCategoryViewState: StateFlow<HomeMovieCategoryViewState> =
+        _upcomingCategoryViewState.asStateFlow()
+    private val _mapper = homeScreenMapper
```

Comment:
No need to create a new value, you can just use `homeScreenMapper` where you use `_mapper`